dataset/format_procedure.py

```
@@ -0,0 +1,70 @@
+# ruff: noqa: T201
+# This script needs to print.
+
+import glob
+import json
+from pathlib import Path
+
+import tiktoken
+from langchain.chat_models import ChatOpenAI
+from langchain.schema import HumanMessage, SystemMessage
+
+
+chatgpt = ChatOpenAI(temperature=0, model_name="gpt-4")
+with Path("./prompt.json").open() as f:
+    prompt = json.load(f)
+context = prompt["context"]
+task = prompt["task"]
+TOKEN__LIMIT = 8192
+
+
+def format_procedure(full_procedure: str) -> str:
+    """Formats a single procedure with GPT-4"""
+    temp = task.format(full_procedure)
+    encoding = tiktoken.encoding_for_model("gpt-4")
+    messages = [SystemMessage(content=context), HumanMessage(content=temp)]
+    if len(encoding.encode(context + temp)) < TOKEN__LIMIT:
+        resp = chatgpt.generate(messages=[messages])
+    else:
+        return ""
```

We should probably throw an error here, then catch the error in the main function, log it, and skip writing anything in the output file.

---

I see wym, but with the changes I made to the code, the writing in the output file now is done in this method.
So there's no need to throw an error to the main method, I'm just logging the files that exceed the token limit.